File: src/Python/367.ValidPerfectSquare.py

class Solution:
    def isPerfectSquare(self, num):
        """
        :type num: int
        :rtype: bool
        """
        left = 1
        right = num // 2 + 1    
        while left <= right:
            middle = (left + right) // 2
            r = middle * middle
            if r == num:
                return True
            elif r <= num:
                left = middle + 1
            else:
                right = middle - 1
        return False
            
if __name__ == '__main__':
    solution = Solution()
    print(solution.isPerfectSquare(1))
    print(solution.isPerfectSquare(16))
    print(solution.isPerfectSquare(34))
else:
    pass

File: src/Java/367.ValidPerfectSquare.java

import java.awt.List;
import java.lang.annotation.Retention;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.Queue;
import java.util.Set;
import java.util.HashSet;
import java.util.HashMap;
import java.util.LinkedList;

class Solution {
    public boolean isPerfectSquare(int num) {
        if (num > 2147395600)
        return false;
        if (num == 2147395600)
            return true;
        float x = 1;
        while (Math.abs(x * x - num) > 0.0001)
            x = x - (x * x - num) / (2 * x);
        return Math.abs((int)(x) - x) < 0.0001;
    }
}


